### PR TITLE
feat(docs): move to vitepress

### PR DESCRIPTION
**Pull request by davidkna**
#### Description

This PR moves the website generation from VuePress (1) to VitePress. This updated config file is now at `docs/.vitepress/config.mts`. I also bumped the bundled nerd font version (handling the NF v3 update) and clarified the font-face definition to only cover PUA code points.

[Preview link](https://shiny-cucurucho-b4b5d4.netlify.app/)

Issues that were handled in the migration:
- `.vuepress` was moved to `.vitepress` and `.vuepress/public` to `public`
- VitePress doesn't treat `README.md` files as index files, to avoid renaming the files I used the `rewrites` options to map the files to what VitePress expects (`index.md`)
- Because internally the files now have a different name, I moved all links from `x/y/z/README.md` to `x/y/z`, avoiding mentioning the filenames directly
- I changed absolute links to relative links to avoid inexpertly have them moving users from translated docs into the English docs
- Some translated files [contained invalid HTML](https://github.com/starship/starship/blob/master/docs/es-ES/config/README.md?plain=1#L2206-L2224), which had to be corrected before the build could proceed. To avoid issues with website publishes in the future due to the stronger checks, I added a CI check that builds the documentation.

#### Motivation and Context

From #5138

> VuePress 1.x is stagnant, lacking development and maintenance. To keep the documentation site modern and accessible, we need to upgrade to the latest version.

This also means the doc builds can be done on recent node versions without issues again. External plug-ins are no longer required for features like sitemap generation, light/dark mode and code copy buttons. Sadly, the light/dark mode toggle depends more strongly on JavaScript compared to the previous version.

Both VuePress 2 and VitePress can be considered successors to VuePress 1, and both don't have a stable release yet and are in the RC phase.

Alternative to #5138

#### How Has This Been Tested?

- [x] I have tested using **MacOS**
- [ ] I have tested using **Linux**
- [ ] I have tested using **Windows**

#### Checklist:

- [x] I have updated the documentation accordingly.
- [x] I have updated the tests accordingly.

### CODE REVIEW

**Review by matchai — APPROVED** [review]
Amazing work! Things look fantastic with a fresh coat of paint.
Thank you for taking this on. It looks like it took a great deal of effort to keep things working as they were.

I've just got some style update suggestions, but I'll make a PR into yours to tackle those.

**Comment by matchai** [comment]
This PR should patch up my suggestions: https://github.com/davidkna/starship/pull/83

File: docs/.vitepress/theme/index.styl [hunk 1]
```diff
@@ -0,0 +1,72 @@
+@import "palette"
```
[the remark this answers is not in the document]
Comment (davidkna) [reply docs/.vitepress/theme/index.styl]: Thanks! I've merged the PR and replaced the stylus with plain CSS.

**Review by andytom — APPROVED** [review]

**Comment by davidkna** [comment]
Thanks for the reviews @matchai and @andytom!

**Pull request merged**